CRM/Financeextras/Hook/BuildForm/RefundButton.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+use CRM_Financeextras_ExtensionUtil as E;
+
+/**
+ * Implement Refund Button if supported Refund by Payment Processor
+ */
+class CRM_Financeextras_Hook_BuildForm_RefundButton {
+
+
+  /**
+   * @var string
+   *   Path where template with javascript stored for adding new Refund Button.
+   */
+  private $templatePath;
+
+  /**
+   * @var \CRM_Contribute_Form_AdditionalPayment
+   *   Form object that is being altered.
+   */
+  private $form;
+
+  /**
+   * CRM_Financeextras_Hook_BuildForm_RefundButton constructor.
+   *
+   * @param \CRM_Contribute_Form_AdditionalPayment $form
+   */
+  public function __construct(&$form) {
+    $this->form = $form;
+    $this->templatePath = E::path() . '/templates/CRM/AdditionalPayment/Form/Payment';
+  }
+
+  /**
+   * Implements changes to form loaded in $this->form
+   */
+  public function buildForm() {
+    $this->addRefundButton();
+  }
+
+  /**
+   * Added Javascript in template to add the button for submit refund
+   */
+  private function addRefundButton() {
```

Comment:
I see that you add a logic that if the contribution is eligible by reading a text "Submit Credit Card Refund", this is quite excessive, you should only attach the link by adding a logic here. 

You should be able to get a contribution ID by using `$form->getVar('_id')` or check the` $form `object  here. By doing this, you should be able to add a unit test here too.